pkgs/applications/networking/cluster/terraform-providers/vpsadmin/default.nix

```diff
@@ -0,0 +1,27 @@
+{ stdenv, fetchFromGitHub, buildGoModule }:
+buildGoModule rec {
+  pname = "terraform-provider-vpsadmin";
+  version = "0.1.0";
+
+  src = fetchFromGitHub {
+    owner = "vpsfreecz";
+    repo = "terraform-provider-vpsadmin";
+    rev = "v${version}";
+    hash = "sha256-+6jRjcManQdoKh7ewOJI1UaulY5OSbkIUHmtrBI33u4=";
+  };
+
+  modSha256 = "sha256-gz+t50uHFj4BQnJg6kOJI/joJVE+usLpVzTqziek2wY=";
```

Comment:
They don't use a `vendor` folder, right?

Reply:
Even if they do, `buildGoModule` does not support invocations without `modSha256` at the moment.

Reply:
Well, we do have somewhat limited support for these kind of terraform modules with `patchGoModVendor` - maybe open an issue upstream and ask if they plan to package their modules in the future too?

Then we could use the generic updater framework we already have. cc @mmilata

Reply:
It's possible to use [nix-prefetch](https://github.com/msteen/nix-prefetch) to get the latest checksum. I use this in [nix-update](https://github.com/Mic92/nix-update/)

Reply:
> They don't use a vendor folder, right?
I wish, otherwise I would have used the other method.

Reply:
There are a number of improvements that we could do around the `update-all` script that are out of scope for this PR but that would nice to have soon:

* avoid re-fetching if the version hasn't change. At the moment `update-all` prefeteches all the repos even if they are already in `data.nix`
* support a "go repo type" as a second argument to the slugs to handle these other cases

Reply:
I agree. I didn't meant to say this is in-scope for this PR.

But it'd be good to at least have an issue opened to track these things.

Reply:
I think it's good to have the discussion. I don't know if an extra issue is going to help given how much issues are open already.

Reply:
I'm not sure if people will find it if it's just here - I'd rather see it somewhere deep in the backlog, than just in our minds ;-)